src/ScheduleFunctions.cpp

```diff
@@ -1020,7 +1023,7 @@ class PrintUsesOfFunc : public IRVisitor {
 
     void visit(const For *op) {
         if (ends_with(op->name, Var::outermost().name()) ||
-            ends_with(op->name, LoopLevel::root().to_string())) {
+            ends_with(op->name, LoopLevel::root().lock().to_string())) {
```

Comment:
It's weird that this mutates a static loop level. At the very least, I would worry about race conditions. I think we should make make the static loop levels locked from the beginning so this isn't necessary.

Reply:
LoopLevel::root() returns a new LoopLevel every time you call it, so it's mutating something different.

If we made these pre-locked, it would prevent you from setting default values that you could override later, e.g.

    LoopLevel w = LoopLevel::root();
    bool weird_condition;
    Func out = compute_something(w, &weird_condition);
    if (weird_condition) w.set(...);